credentials/credentials.go

```diff
@@ -190,16 +196,16 @@ func NewTLS(c *tls.Config) TransportCredentials {
 }
 
 // NewClientTLSFromCert constructs a TLS from the input certificate for client.
-// serverNameOverwrite is for testing only. If set to a non empty string,
-// it will overwrite the virtual host name of authority (e.g. :authority header field) in requests.
-func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverwrite string) TransportCredentials {
-	return NewTLS(&tls.Config{ServerName: serverNameOverwrite, RootCAs: cp})
+// serverNameOverride is for testing only. If set to a non empty string,
+// it will override the virtual host name of authority (e.g. :authority header field) in requests.
+func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverride string) TransportCredentials {
```

Comment:
I'm still do not understand why you need this, since you can set any tls option you want using the `NewTLS` function:

```go
credentials.NewTLS(&tls.Config{ServerName: "any server name"})
```